fdbserver/DataDistribution.actor.cpp

```diff
@@ -3106,7 +3110,20 @@ ACTOR Future<Void> teamTracker(DDTeamCollection* self, Reference<TCTeamInfo> tea
 			ASSERT( self->optimalTeamCount >= 0 );
 			self->zeroOptimalTeams.set(self->optimalTeamCount == 0);
 		}
-		throw;
+		try {
+			bool containsFailed = teamContainsFailedServer(self, team);
+			// If server is marked as failed, a lost race condition could result in RelocateShards not being sent
+			// This is implemented as a safety net so we do not have orphaned shards sitting around
+			if (containsFailed && e.code() == error_code_actor_cancelled) {
+				TraceEvent("TeamTrackerStoppedSendRelocateToDDQ", self->distributorId).detail("Team", team->getDesc());
+				teamTrackerSendRelocateShards(self, team, redundantTeam, containsFailed, serversLeft);
+			}
+			TraceEvent("TeamTrackerStopped", self->distributorId).error(e);
+			throw e;
+		} catch (Error& err) {
+			TraceEvent("TeamTrackerStoppedSendRelocateError", self->distributorId).error(err);
```

Comment:
Can you make this a SevError event so that we are confident this won't happen?

Reply:
I added this in my local branch and no error was found in 100K correctness tests.

Reply:
This won't work, will it? Because the `throw e` directly above will always be caught, then we are always logging a sev40. This was one of the things I had trouble with when ordering the throws and try-catches

Reply:
Ah, I remember that `.error` on a trace event will not apply if the error code is `error_code_actor_cancelled` so theoretically that could save us here. Let me push a few changes and let me know if that's the behaviour we want

Reply:
I think your recent update is a better solution. 
I also need to run more tests to ensure this buggy scenario is triggered.